pkg/adapter/awssnssource/adapter.go

```diff
@@ -129,22 +113,20 @@ func (a *adapter) Start(ctx context.Context) error {
 		if serverErr != nil {
 			err = fmt.Errorf("failure during runtime of SNS notification handler: %w", serverErr)
 		}
-		cancel()
 
 	case <-ctx.Done():
 		a.logger.Info("Shutting server down")
 
-		shutdownCtx, cancelTimeout := context.WithTimeout(ctx, serverShutdownGracePeriod)
-		defer cancelTimeout()
-		if shutdownErr := server.Shutdown(shutdownCtx); shutdownErr != nil {
+		ctx, cancel := context.WithTimeout(context.Background(), serverShutdownGracePeriod)
```

Comment:
As explained in the referenced issue, `ctx` has already expired at that point, so `WithTimeout()` needs to be based on a fresh context, otherwise it will return immediately instead of waiting for a graceful shutdown.